precision selection for PIConGPU quantities

psychocoderHPC opened the pull request:
Allow choosing the precision for PIC quantities.
By default, the user-selected precision for PIC quantities.

chillenzer commented on include/picongpu/unitless/simulation.unitless:
```diff
@@ -188,13 +197,14 @@ namespace picongpu
             constexpr float_64 typicalNumParticlesPerMacroParticle() const
             {
                 return (si.getBaseDensity() * si.getCellSize().productOfComponents())
-                    / float_64(getTypicalNumParticlesPerCell());
+                    / getTypicalNumParticlesPerCell<float_64>();
             }
         };
```
Why's this hard-coded to `float_64`?

psychocoderHPC replied:
To avoid compiler warnings, we cast from an integer type to floating point type. It was an explicit cast before too.

chillenzer reviewed (dismissed):
Hope you didn't do all those changes by hand. Happy when the CI is.

psychocoderHPC commented:
> Hope you didn't do all those changes by hand. Happy when the CI is.

For sure I made the changes by hand.

psychocoderHPC committed:
￼precision selection for PIConGPU quantities

Allow choosing the precision for PIC quantities.
By default, the user-selected precision for PIC quantities.

ci: picongpu
